src/app/containers/PageHandlers/withVariant/index.test.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { useParams, useLocation } from 'react-router-dom';
+import WithVariant from '.';
+import { frontpageRegexPath } from '#app/routes/regex';
+
+jest.mock('react-router-dom', () => ({
+  useParams: jest.fn(),
+  useLocation: jest.fn(),
+  // eslint-disable-next-line react/prop-types
+  Redirect: ({ to: { pathname } }) => <p>You are going to {pathname}</p>,
+}));
+
+describe('WithVariant', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const Component = () => <h1>This is the BBC.</h1>;
+  const WithVariantHOC = WithVariant(Component);
+
+  const getMatchProps = (service, path = null) => ({
+    path: path || frontpageRegexPath,
+    params: {
+      service,
+    },
+  });
+
+  describe('service with no default variant', () => {
+    beforeEach(() => {
+      const service = 'news';
+
+      useParams.mockReturnValue({
+        service,
+      });
+      useLocation.mockReturnValue({
+        pathname: `/${service}`,
+      });
+    });
+
+    const match = getMatchProps('news');
+
+    shouldMatchSnapshot(
+      'should not redirect',
+      <WithVariantHOC match={match} />,
+    );
+  });
+
+  describe('service (ukchina) with default variant', () => {
+    beforeEach(() => {
+      const service = 'ukchina';
+
+      useParams.mockReturnValue({
+        service,
+      });
+      useLocation.mockReturnValue({
+        pathname: `/${service}`,
+      });
+    });
+
+    const match = getMatchProps('ukchina');
+
+    shouldMatchSnapshot(
```

Comment:
snapshots don't seem like the right kind of test here. What about something like https://medium.com/@the_teacher/how-to-test-redirect-from-react-router-with-rtl-react-test-library-and-jest-242eced1c6b8

Reply:
I'll modify to assert renders in-memory.

Reply:
The `Redirect` component of `react-router-dom` is mocked to be:

```jsx
({ to: { pathname } }) => <p>You are going to {pathname}</p>
```

If a redirect occurs, we'll know from the rendered output.  Why aren't snapshots right for these tests?

Reply:
I think it's a lot clearer to have something like `expect(getByText('You are going to {pathname}'))toBeTruthy()`

Reply:
There's docs on how to test redirects without mocking the redirect component https://reacttraining.com/react-router/web/guides/testing I was able to do this quite easily https://github.com/bbc/simorgh/pull/4643/files#diff-38e819f9fe128b91a0252322e174ab51

Reply:
I'll stay away from mocking in the future, as much as I can. Thanks for the links.